src/mqtt/client.rs

```diff
@@ -80,8 +80,8 @@ pub struct MqttClientConfiguration<'a> {
     pub client_certificate: Option<X509<'static>>,
     pub private_key: Option<X509<'static>>,
     pub private_key_password: Option<&'a str>,
-    // TODO: Future
-    // pub psk_hint_key: KeyHint,
+
+    pub psk: Option<Psk<'a>>,
```

Comment:
Protect with `#[cfg(esp_idf_esp_tls_psk_verification)]`. This way the user **must** enable this config option to use the functionality. The fact that ESP IDF requires it only at runtime is bad design IMO, that we should not follow.

Reply:
This looks like the thing I was looking for in my last commit. I like this way of being on the same page in esp-idf-svc and esp-idf about this feature gate. :)

Reply:
Used `#[cfg(all(esp_idf_esp_tls_psk_verification, feature = "alloc"))]`